core/pbcc_request_retry_timer.c

```diff
@@ -0,0 +1,198 @@
+/* -*- c-file-style:"stroustrup"; indent-tabs-mode: nil -*- */
+#include "pbcc_request_retry_timer.h"
+
+#include <pthread.h>
+#include <stdbool.h>
+#include <unistd.h>
+
+#include "lib/msstopwatch/msstopwatch.h"
+#include "core/pbcc_memory_utils.h"
+#include "core/pubnub_assert.h"
+#include "core/pubnub_mutex.h"
+#include "core/pb_sleep_ms.h"
+#include "pubnub_internal.h"
+
+
+// ----------------------------------------------
+//                   Constants
+// ----------------------------------------------
+
+/**
+ * @brief Polling timeout.
+ *
+ * Timeout used with sleep function inside of `while` loop to break constant
+ * polling and give other processes some CPU time.
+ */
+#define POLLING_TIMEOUT 10
```

Comment:
Risky define. 
If anyone would import that we can redefine their definitions. 
```suggestion
#define PBCC_POLLING_TIMEOUT 10
```